haskell/assets/ghci_script

```diff
@@ -3,6 +3,11 @@
 import qualified GHC.IO.Handle as Handle
 import qualified System.IO as IO
 import qualified System.Directory as Dir
+:set -Wno-name-shadowing
```

Comment:
Isn't this `:set` kept even after reload?

Reply:
That is correct. I've changed it to use `:seti` instead of `:set`. That way any shadowing in source files will still cause a warning while the warning will only be disabled in the interactive session itself.

I haven't found a way to avoid affecting that warning altogether. `ghci_script` requires bind syntax, so we have to bind `(>>=)` and friends to the `Prelude`'s definitions, this requires `-Wno-name-shadowing` to work if the module being loaded defines different bindings for `(>>=)` etc. I haven't found a way to only temporarily `:seti`.

Reply:
A solution would be to not use the do-notation in the script, but explicitely call `Prelude.>>=` everywhere. But that's quite painful

Reply:
I've tried that but sadly it's not enough. We need to bind the outcome of some IO operations using bind syntax in the REPL directly, e.g. 
```
rules_haskell_stdout_dupe <- Handle.hDuplicate IO.stdout
```
That implicitly requires `do` notation.